library/src/hooks/street-view-panorama.ts

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable complexity */
+import {useEffect, useState} from 'react';
+
+import {useGoogleMap} from './map-instance';
+
+export interface StreetViewPanoramaProps {
+  divElement?: HTMLElement | null;
+  position?: google.maps.LatLng | google.maps.LatLngLiteral;
+  pov?: google.maps.StreetViewPov;
+  zoom?: number;
+}
+
+/**
+ * Hook to get Street View Panorama
+ */
+export const useStreetViewPanorama = (
+  props: StreetViewPanoramaProps
+): google.maps.StreetViewPanorama | null => {
+  const {divElement, position, pov, zoom} = props;
+
+  // Get map instance
+  const {map} = useGoogleMap();
+
+  const [streetViewPanorama, setStreetViewPanorama] =
+    useState<google.maps.StreetViewPanorama | null>(null);
+
+  // Creates a Street View instance
+  useEffect(() => {
+    // Wait for Google Maps API to be initialized
+    if (!map) {
+      return (): void => {};
+    }
```

Comment:
The map variable shouldn't be the one who indicates whether the API is loaded or not. Should I maybe re-add the loading state in #93? I thought we never used it, but I think this would be the case here. What do you think?

Reply:
Hm... We basically have this check in all hooks and I think we need it to have the map instance ready. So, I can update the comment for this line, if you prefer? (Because I will need the map in line 64 and 70 for example)

What would be the advantage of a loading state here at this place?

Reply:
If you use a Street View Panorama within its own container element, you don't need a map instance. It would be possible to just load the Google Maps API (without creating a map instance) and create the Street View Panorama when the API is loaded.

Reply:
Okay, then, so let's remove `map.setStreetView(newPanorama);` in line 60 and instead we only check for the map in line 36 (where no dive element is present, but we need a map instance) and I wait for your PR [#93 ](https://github.com/ubilabs/google-maps-react-hooks/pull/93) to be able to check if only the api is loaded in line 53 (div element is present).

This means, we will leave it up to the user/developer to connect the div to a map in this case and I will move line 60 into the example case 👍  (and probably documentation)